pennylane/workflow/qnode.py

```diff
@@ -985,9 +805,9 @@ def construct(self, args, kwargs):  # pylint: disable=too-many-branches
         old_interface = self.interface
 
         if self._qfunc_uses_shots_arg:
-            shots = _get_device_shots(self._original_device)
+            shots = self._original_device.shots

```

Comment:
Since no more device switching in the qnode, we can get rid of `QNode._original_device` and just use `self.device` everywhere.